src/raps/unlockAndDepositUniswap.ts

```diff
@@ -0,0 +1,100 @@
+import { concat, reduce } from 'lodash';
+import {
+  assetNeedsUnlocking,
+  estimateApprove,
+  estimateDepositUniswap,
+} from './actions';
+import {
+  createNewAction,
+  createNewRap,
+  RapAction,
+  RapActionTypes,
+  SwapActionParameters,
+} from './common';
+import store from '@rainbow-me/redux/store';
+import { ethUnits, ZapInAddress } from '@rainbow-me/references';
+import { add } from '@rainbow-me/utilities';
+
+export const estimateUnlockAndDepositUniswap = async (
+  swapParameters: SwapActionParameters
+) => {
+  const { inputAmount } = swapParameters;
+  const { inputCurrency } = store.getState().swap;
+
+  if (!inputCurrency || !inputAmount) return ethUnits.basic_deposit_uniswap;
+
+  const {
+    accountAddress,
+  }: {
+    accountAddress: string;
+  } = store.getState().settings;
+
+  let gasLimits: (string | number)[] = [];
+  const depositAssetNeedsUnlocking = await assetNeedsUnlocking(
+    accountAddress,
+    inputAmount,
+    inputCurrency,
+    ZapInAddress
+  );
+  if (depositAssetNeedsUnlocking) {
+    const unlockGasLimit = await estimateApprove(
+      accountAddress,
+      inputCurrency.address,
+      ZapInAddress
+    );
+    gasLimits = concat(
+      gasLimits,
+      unlockGasLimit,
+      ethUnits.basic_deposit_uniswap
+    );
+  } else {
+    const depositGasLimit = await estimateDepositUniswap(inputAmount);
+    gasLimits = concat(gasLimits, depositGasLimit);
+  }
+
+  return reduce(gasLimits, (acc, limit) => add(acc, limit), '0');
+};
+
+export const createDepositUniswapRap = async (
+  swapParameters: SwapActionParameters
+) => {
+  const { inputAmount } = swapParameters;
+  const { inputCurrency } = store.getState().swap;
+  const {
+    accountAddress,
+  }: {
+    accountAddress: string;
+  } = store.getState().settings;
+
+  let actions: RapAction[] = [];
+
+  const depositAssetNeedsUnlocking = await assetNeedsUnlocking(
+    accountAddress,
+    inputAmount,
+    inputCurrency,
+    ZapInAddress
```

Comment:
I think we have all the const that are addresses using all uppercase chars, which is something that I personally in favor of.